README.md

```diff
@@ -1 +1,199 @@
-# install-esp-idf-action
\ No newline at end of file
+# ESP-IDF Installation Action
+
+This GitHub Action automates the installation of the ESP-IDF framework on GitHub-hosted runners. It supports Windows, macOS, and Linux platforms, allowing you to set up ESP-IDF for your CI/CD workflows.
+
+## Features
+
+- Cross-platform support (Windows, macOS, Linux) thanks to [EIM](https://github.com/espressif/idf-im-cli)
+- Automatic installation of required system dependencies
+- Configurable ESP-IDF version and installation paths
+- Sets up all necessary environment variables and tools
+- Adds ESP-IDF commands (`idf.py`, `esptool.py`, etc.) to PATH
+
+## Usage
+
+Basic usage with default settings:
+
+```yaml
+steps:
+  - uses: actions/checkout@v4
+  - name: Install ESP-IDF
+    uses: espressif/esp-idf-install-action@v1
+  - name: Build your project
+    run: |
+      idf.py build
+```
+
+Advanced usage with custom configuration:
+
+```yaml
+steps:
+  - uses: actions/checkout@v4
+  - name: Install ESP-IDF
+    uses: espressif/esp-idf-install-action@v1
+    with:
+      esp_idf_version: "v5.0"
+      esp_idf_path: "/custom/path/to/esp-idf"
+      esp_idf_tools_path: "/custom/path/to/tools"
```

Comment:
Nitpick: Since the arguments are passed to "espressif/esp-idf-...", I think prefixing everything with `esp_idf_` might not be necessary.

Btw, looking at actions maintained by Github (e.g. https://github.com/actions/checkout), seems like using hyphens in argument names is more common now (i.e. `tools-path` rather than `tools_path`).